packages/@react-aria/menu/src/useMenu.ts

```diff
@@ -68,15 +76,78 @@ export function useMenu<T>(props: AriaMenuOptions<T>, state: TreeState<T>, ref:
     linkBehavior: 'override'
   });
 
+  let id = useId(props.id);
   menuData.set(state, {
     onClose: props.onClose,
-    onAction: props.onAction
+    onAction: props.onAction,
+    shouldUseVirtualFocus: props.shouldUseVirtualFocus,
+    onHoverStart,
+    id
   });
 
+  // TODO: for now I'm putting these announcement into menu but would like to discuss where we may this these could go
+  // I was thinking perhaps in @react-aria/interactions (it is interaction specific aka virtualFocus) or @react-aria/collections (dependent on tracking a focused key in a collection)
+  // but those felt kinda iffy. A new package?
+  // TODO: port all other translations/remove stuff if not needed
```

Comment:
I'd feel more comfortable having them in a separate package since these announcements will have to happen in a for any autocomplete wrapped collection element. Open to opinions